pywwt/annotation.py

```diff
@@ -252,6 +255,100 @@ def _on_trait_change(self, changed):
         super(Line, self)._on_trait_change(changed)
 
 
+class FieldOfView():
+    """
+    A collection of polygon annotations. Takes the name of a pre-loaded 
+    telescope and displays its field of view.
+    """
+    # links for kepler. guide: https://keplerscience.arc.nasa.gov/k2-fields.html
+    # footprint: https://raw.githubusercontent.com/KeplerGO/K2FootprintFiles/master/json/k2-footprint.json
+
+    # more efficient method than CircleCollection of changing trait values?
+    def __init__(self, parent, telescope, center, rot, **kwargs):
+        self.parent = parent
+        self.available = self.parent.instruments.available
+        self.dim = self.parent.instruments.dim
+        
+        # list of IDs of annotations created in this FieldofView instance
+        self.active = []
+        self._gen_fov(telescope, center, rot, **kwargs)
+
+    def _gen_fov(self, telescope, center, rot, **kwargs):
+        #telescope = telescope.lower()
+        if telescope not in self.available:
+            raise ValueError('the given telescope\'s field of view is unavailable at this time')
+
+        position = self.dim[telescope][0]
+        dimensions = self.dim[telescope][-1]
+        # dimensions is a list of lists [ [[ras], [decs]], '', ...]
+
+        # test that pos matches what the user entered
+        if center and position == 'absolute':
+            raise ValueError('the given telescope does not take center coordinates')
+        elif not center and position == 'relative':
+            raise ValueError('the given telescope requires center coordinates')
+
+        # draw jwst nircam's inner panels first
+        if telescope[-6:] == 'nircam':
+            self._gen_fov('nircam_short', center, rot, **kwargs)
+
+        for panel in dimensions:
+            ras = (panel[0] * u.deg).value
+            decs = (panel[1] * u.deg).value
+            ra_r, dec_r = self._rotate(ras, decs, rot)
+
+            if telescope == 'k2':
+                ra_tr = ra_r; dec_tr = dec_r
+            else:
+                center_ra = center.ra.to(u.deg).value
+                center_dec = center.dec.to(u.deg).value
+
+                dec_tr = [center_dec + dec for dec in dec_r]
+                # scale RA by dec (due to polar contraction of spherical coords)
+                ra_tr = [center_ra +
+                         ra / np.cos((dec_tr[i] * u.deg).to(u.rad)).value
+                         for i, ra in enumerate(ra_r)]
```

Comment:
Likewise, can you not write ``ra_tr = center_ra + ra / np.cos((dec_tr[i] * u.deg).to(u.rad)).value``?

Note that the ``to(u.rad)`` is actually not needed, as ``np.cos`` will correctly recognize the degree unit

Reply:
The same comment from above on needing lists and thus list comprehensions applies here.

For `np.cos`, I see now that degrees work. The documentation says radians only, but I guess that's not the case when the argument is an Astropy quantity.